### PR TITLE
[TwigBridge] Foundation form layout integration

### PR DESCRIPTION
**Pull request by totophe**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| License | MIT |

As symfony integrate in its roots Bootstrap layout, why not also include Foundation layout? I think it's not a matters of including all the css frameworks in symfony but at least having two instead of one would be interesting to have a basic choice.

@hhamon told me I should create this PR :-D :-p ;-)

### CODE REVIEW

**Comment by fabpot** [comment]
It's also a matter of maintaining the theme. Any @symfony/deciders willing to maintain it. Personally, I'm :-1:

**Comment by Seldaek** [comment]
I'd :+1: as I believe Foundation is a better base for custom apps (more neutral default theme, less override rules needed in the CSS to achieve a custom look) while Bootstrap is better geared at rapid prototyping and I'd argue Symfony is probably still better at building sustainable custom apps than prototypes.

File: src/Symfony/Bridge/Twig/Resources/views/Form/foundation_layout.html.twig [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        {{- block('form_widget_simple') -}}
+    {% else %}
+        {% set attr = attr|merge({class: (attr.class|default('') ~ ' row')|trim}) %}
+        {% if datetime is not defined or not datetime %}
+            <div {{ block('widget_container_attributes') -}}>
+        {% endif %}
+        {{ date_pattern|replace({
+        '{{ year }}': '<div class="large-4 columns">' ~ form_widget(form.year) ~ '</div>',
+        '{{ month }}': '<div class="large-4 columns">' ~ form_widget(form.month) ~ '</div>',
+        '{{ day }}': '<div class="large-4 columns">' ~ form_widget(form.day) ~ '</div>',
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Resources/views/Form/foundation_layout.html.twig]: wrong indentation here

**Comment by stof** [comment]
The whitespace control is missing in this template to remove extra spaces (as done in the default theme)

**Comment by fabpot** [comment]
@totophe Any news on this PR? Are you willing to finish it?

**Comment by totophe** [comment]
Yes, sorry, I was drawn into our sprints. I'll do that ASAP.

**Comment by totophe** [comment]
Hi, changes are made. Sorry @stof for the delay. I'll not procrastinate next time.

**Comment by Reyncor** [comment]
Would have saved me a some time had this been in the Twig bundle... Thank you!

File: src/Symfony/Bridge/Twig/Resources/views/Form/foundation_layout.html.twig [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+{% block form_errors -%}
+    {% if errors|length > 0 -%}
+        {% if form.parent %}<small class="error">{% else %}<div data-alert class="alert-box alert">{% endif %}
+        {%- for error in errors -%}
+            {{ error.message }}
+            {% if not loop.last %}, {% endif %}
+        {%- endfor -%}
+        {% if form.parent %}</small>{% else %}</div>{% endif %}
+    {%- endif %}
+{%- endblock form_errors %}
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Resources/views/Form/foundation_layout.html.twig]: Missing EOL

**Comment by nicolas-grekas** [comment]
@totophe could you open a doc PR (see this one for the bootstrap theme: https://github.com/symfony/symfony-docs/pull/4411)

**Comment by nicolas-grekas** [comment]
:+1: with some comments above

**Comment by totophe** [comment]
@nicolas-grekas I'll fix that asap (this week)

**Comment by nicolas-grekas** [comment]
ping @totophe

File: src/Symfony/Bridge/Twig/Resources/views/Form/foundation_layout.html.twig (hunk 2 above)
Comment (totophe) [inline src/Symfony/Bridge/Twig/Resources/views/Form/foundation_layout.html.twig]: Don't get it, where?

**Comment by totophe** [comment]
Doc PR done: https://github.com/symfony/symfony-docs/pull/5702 :-)

File: src/Symfony/Bridge/Twig/Resources/views/Form/foundation_layout.html.twig (hunk 2 above)
Comment (totophe) [inline src/Symfony/Bridge/Twig/Resources/views/Form/foundation_layout.html.twig]: Fixed

**Comment by javiereguiluz** [comment]
To be consistent with our Bootstrap practice and to be future-proof, we should probably rename the Twig template to `foundation_5_layout.html.twig`

**Comment by totophe** [comment]
Here you go

**Comment by florianbelhomme** [comment]
I'd :+1: Foundation make developement easier.

**Comment by fabpot** [comment]
ping @symfony/deciders

**Comment by dunglas** [comment]
Foundation is interesting and it looks like a good idea to have support for it in Symfony but as I don't often use it and will not help to maintain this theme I prefer abstention.

**Comment by xabbuh** [comment]
I am neither pro nor contra. The implementation looks good, but it would be good to have someone being able to maintain the theme. On the other hand, a form theme is not a real big deal and could probably live in the core nonetheless.

**Comment by fabpot** [comment]
ok, let's merge it in 2.8. If we have too many bug reports and noone fixing them, we will still be able to deprecate it and remove it later on.

**Comment by fabpot** [comment]
Thank you @totophe.

**Comment by totophe** [comment]
As I've proposed this feature, I'm willing to maintain it if any issue would occur.

**Pull request closed without merging**